lib/web_ui/dev/goldens.dart

```diff
@@ -140,3 +145,85 @@ class ImageDiff {
 String getPrintableDiffFilesInfo(double diffRate, double maxRate) =>
   '(${((diffRate) * 100).toStringAsFixed(4)}% of pixels were different. '
   'Maximum allowed rate is: ${(maxRate * 100).toStringAsFixed(4)}%).';
+
+/// Fetches golden files from github.com/flutter/goldens, cloning the repository if necessary.
+///
+/// The repository is cloned into web_ui/.dart_tool.
+Future<void> fetchGoldens() async {
+  await _GoldensRepoFetcher().fetch();
+}
+
+class _GoldensRepoFetcher {
+  String _repository;
+  String _revision;
+
+  Future<void> fetch() async {
+    final io.File lockFile = io.File(
+      path.join(environment.webUiDevDir.path, 'goldens_lock.yaml')
+    );
+    final YamlMap lock = loadYaml(lockFile.readAsStringSync());
+    _repository = lock['repository'];
+    _revision = lock['revision'];
+
+    final String localRevision = await _getLocalRevision();
+    if (localRevision == _revision) {
+      return;
+    }
+
+    print('Local revision $localRevision is different from $_revision in goldens_lock.yaml.');
```

Comment:
Will this reset hard my goldens repository if I run the tests "one more time" after comitting the goldens, but before sending the PR? :P

Reply:
Cirrus failing with:

```
Local revision null is different from dd993a32c23c5c542f083134467e7cda09cac975 in goldens_lock.yaml.
Will fetch the requested revision.
Initialized empty Git repository in /tmp/clean_engine/src/flutter/lib/web_ui/.dart_tool/goldens/.git/
error: no such remote ref dd993a32c23c5c542f083134467e7cda09cac975
```
However ref [exists](https://github.com/flutter/goldens/commit/dd993a32c23c5c542f083134467e7cda09cac975) (wrong origin?)

Reply:
> Will this reset hard my goldens repository if I run the tests "one more time" after comitting the goldens, but before sending the PR?

Hmm, yes it will :)

Reply:
Fixed

Reply:
I'm not going to try to make too robust. I'd rather invest my energy into Skia Gold integration.

Reply:
Sounds good!